stats/kafka/collector.go

```diff
@@ -0,0 +1,173 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package kafka
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/stats"
+	"github.com/loadimpact/k6/stats/influxdb"
+	jsonc "github.com/loadimpact/k6/stats/json"
+	log "github.com/sirupsen/logrus"
+)
+
+// Collector implements the lib.Collector interface and should be used only for testing
+type Collector struct {
+	Producer sarama.SyncProducer
+	Config   Config
+
+	Samples []stats.Sample
+	lock    sync.Mutex
+}
+
+// New creates an instance of the collector
+func New(conf Config) (*Collector, error) {
+	producer, err := sarama.NewSyncProducer(conf.Brokers, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Collector{
+		Producer: producer,
+		Config:   conf,
+	}, nil
+}
+
+// Init does nothing, it's only included to satisfy the lib.Collector interface
+func (c *Collector) Init() error { return nil }
+
+// Run just blocks until the context is done
+func (c *Collector) Run(ctx context.Context) {
+	log.Debug("Kafka: Running!")
+	d, err := time.ParseDuration(c.Config.PushInterval.String())
```

Comment:
I think this should be `c.Config.PushInterval.Duration`, there's no reason to serialize the `PushInterval` to string only to parse it back to duration again. Besides, validating options shouldn't be done in `Run()`, `New()` or `Init()` is much more appropriate place for that.

Reply:
will work on this